### PR TITLE
fix(#86): fix validator and update structure

### LINKED ISSUE #86: Validation errors should have the correct field name
When responding with an `application/problem+json` having the `validationErrors` field, `field` should have the correct fieldname that the API data model defines, not the internal Go one:

Example:
```json
{
  "title": "can't create Software",
  "detail": "invalid format",
  "status": 422,
  "validationErrors": [
    {
      "field": "Software.URLs",
      "rule": "required",
      "providedValue": ""
    },
    {
      "field": "Software.PubliccodeYml",
      "rule": "required",
      "providedValue": ""
    }
  ]
}
```

`field` should be `urls` and `publiccodeYml`, not `Software.URLs` and `Software.PubliccodeYml`.

### PR DESCRIPTION
**Pull request by LorenzoS92**
Fix #86 
Updated validator and structure to return proper errors

### CODE REVIEW

**Review by bfabio — CHANGES_REQUESTED** [review]
Great, can we also add the tests for it, with various input?

**Comment by LorenzoS92** [comment]
OMG forgotten it, of course I will do that 🙊

File: main_test.go [hunk 1]
```diff
@@ -195,6 +195,17 @@ func TestPublishersEndpoints(t *testing.T) {
 			},
 			expectedContentType: "application/json",
 		},
+		{
+			query: "POST /v1/publishers - INVALID PAYLOAD",
```
Comment (bfabio) [inline main_test.go:199]:
```suggestion
			description: "POST publishers with invalid payload",
			query: "POST /v1/publishers",
```

This is tricky and works almost by chance (we split at space and use the first two strings for the query), but I'd say we should always provide a `description` when we want add some more information to the test.

It's not ideal and we should probably validate `query` more strictly.

File: main_test.go [hunk 2]
```diff
@@ -195,6 +195,17 @@ func TestPublishersEndpoints(t *testing.T) {
...
+			body:  `{"URL":"https://www.example.com", "email":"error"}`,
+			headers: map[string][]string{
+				"Authorization": {goodToken},
+				"Content-Type":  {"application/json"},
+			},
+			expectedCode:        422,
+			expectedBody:        `{"title":"can't create Publisher","detail":"invalid json","status":422,"validationErrors":[{"field":"email","rule":"email","value":"error"}]}`,
```
Comment (bfabio) [inline main_test.go:206]: This should say `invalid format` to be consistent with the other resources, I'll add a remark at https://github.com/italia/developers-italia-api/issues/12

File: main_test.go [hunk 3]
```diff
@@ -468,6 +479,17 @@ func TestSoftwareEndpoints(t *testing.T) {
 				// TODO: check there are no dangling software_urls
 			},
 		},
+		{
+			query: "POST /v1/software - wrong payload",
```
Comment (bfabio) [inline main_test.go:483]:
```suggestion
			description: "POST software with invalid payload",
			query: "POST /v1/software",
```

**Review by bfabio — CHANGES_REQUESTED** [review]
Can we test all the resources, so all the tests are uniform (Logs as well)?

**Review by bfabio — APPROVED** [review]

**Pull request merged**